### PR TITLE
[BD-46] chore: remove airbnb-prop-types dependency and upgrade eslint in docs project

### PR DESCRIPTION
**Pull request by viktorrusakov**
## Description

Remove `airbnb-prop-types` dependency as it does not support React 17 (see [this issue](https://github.com/airbnb/prop-types/issues/73)), which caused build issues after node upgrade. 
We only used 2 prop types from that package anyway, so it should't be a big change.

### Deploy Preview

Include a direct link to your changes in this PR's deploy preview here (e.g., a specific component page).

## Merge Checklist

* [x] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [x] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [x] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [x] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [x] Were your changes tested in the `example` app?
* [x] Is there adequate test coverage for your changes?
* [x] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @viktorrusakov! I've created [BLENDED-1312](https://openedx.atlassian.net/browse/BLENDED-1312) to keep track of it in Jira.   More details are on the [BD-46](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3134685276/BD-46+Paragon+Blended+Support+and+Maintenance) project page.


When this pull request is ready, tag your edX technical lead.

File: www/package.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
   "devDependencies": {
     "babel-eslint": "^10.0.1",
     "css": "^3.0.0",
-    "eslint": "^7.6.0",
```
Comment (viktorrusakov) [inline www/package.json:59]: this one also cause warnings about peer dependencies, to be honest not sure why we need separate `eslint` for docs site..

File: www/src/components/PropsTable.jsx [hunk 2]
```diff
@@ -28,7 +28,7 @@ const Prop = ({
   <li className="px-4 border-top border-light-300">
     <div className="my-3">
       <div className="mb-2">
-        <span class="mr-2 font-weight-bold">{`${name} `}</span>
+        <span className="mr-2 font-weight-bold">{`${name} `}</span>
```
Comment (adamstankiewicz) [inline www/src/components/PropsTable.jsx:31]: Good catch

File: www/package.json (hunk 1 above)
Comment (adamstankiewicz) [reply www/package.json:59]: Ha, yeah... I guess `www` could have its own eslint config defined for the folder, but could still inherit from the parent `eslint` from the root `node_modules`.

**Review by adamstankiewicz — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@viktorrusakov 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/paragon&entry.867055334=https://github.com/openedx/paragon/pull/1367&entry.1484655318=https://github.com/viktorrusakov&entry.752974735=2022-06-10+08:52&entry.1917517419=2022-06-10+12:36&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**